### ISSUE TITLE
Arms Warrior Bugs

### ISSUE THREAD

**Issue by pnmueller** [simulationcraft/simc#2774]
Edit 2:
I'm assuming the "Raider's Training Dummy" has less armour than the "Custom" target type in Simulationcraft? Otherwise Simulationcraft is consistently showing ~5% less damage for abilities and auto attacks than in-game as Arms.

However, I noticed when comparing damage distribution and subsequently enabling log files and going through the logs that Mortal Strike cooldown seems not to be properly simulated. In the results I can see an interval on MS of 6.16 seconds even though the base cooldown is already at 4.91 seconds (because of haste) on my character, Tactician resets MS cooldown somewhat frequently, MS is set at a very high priority and I disabled any rage usage to below 30 rage (so it shouldn't be delayed due to rage). Here's some relevant parts of the log and APL that also show this issue (I cut everything that's not important):
2.175 Toughone_Active mortal_strike hits Fluffy_Pillow for 74317 physical damage (crit)
3.121 Toughone_Active colossus_smash hits Fluffy_Pillow for 66795 physical damage (crit)
4.067 Toughone_Active slam hits Fluffy_Pillow for 51369 physical damage (crit)
5.012 Toughone_Active slam hits Fluffy_Pillow for 25684 physical damage (hit)
5.957 Toughone_Active slam hits Fluffy_Pillow for 51369 physical damage (crit)
6.903 Toughone_Active overpower hits Fluffy_Pillow for 85226 physical damage (crit)
7.849 Toughone_Active slam hits Fluffy_Pillow for 25684 physical damage (hit)
8.795 Toughone_Active mortal_strike hits Fluffy_Pillow for 37158 physical damage (hit)

MS should've been ready again at 2.175 + 4.91 = 7.085 seconds (or earlier if BL was active at that point already) and yet the simulator opts to use slam at 7.849 seconds - an ability that costs the same amount of rage and is below MS in the action priority list (see default APL). This clearly indicates that the cooldown isn't properly represented.

Edit 3:
Confirmed my suspicion. MS cooldown isn't properly affected by haste. By cutting all actions aside from auto attacks and MS from the APL, I got the following results:
  Actions:
    auto_attack_mh       Count= 56.0|  2.73sec  DPE= 7365|44%  DPET= 2760  DPR=   0.0  pDPS=2749  Miss=0.00%  Hit=5734|5316|6864  Crit=11472|10631|13727|28.4%
    **mortal_strike        Count= 21.0|  6.00sec**  DPE=22587|50%  DPET=19677  DPR=1129.4  pDPS=3162  Miss=0.00%  Hit=17691|16431|21071  Crit=35345|32863|42143|27.7%
    opportunity_strikes  Count=  5.1| 15.87sec  DPE=10903| 6%  DPET=    0  DPR=   0.0  pDPS= 372  Miss=0.00%  Hit=8509|8040|10310  Crit=17017|16080|20620|28.1%

Mortal Strike is used exactly once every 6 seconds which is the base CD of MS (before haste).

Edit 4:
Just noticed that the previous log shows another bug in the simulator: If you check in-game, Tactician can actually proc from MS itself (since it spends rage) and reset its own CD so you'd be able to MS twice in a row. That doesn't seem to be reflected in the simulator as it'd obviously result in less than 6 seconds for the average interval between two executions of MS.

tl;dr: MS cooldown isn't properly reduced by haste. Tactician doesn't properly reset MS cooldown when procced by MS itself (or possibly at all times).

**Comment by Collisionc** [comment]
Eugh. MS resetting itself is going to be a pain in the ass because by default simc consumes resources before triggering cooldowns. I can work around it, tho.

**Comment by Collisionc** [comment]
Also, make sure you are not using the target dummy outside your garrison, because for whatever insane reason it takes 5% more damage than it should.

**Comment by Collisionc** [comment]
I did that in kind of a rush, but it looks like it's working. I'll upload a new version.

**Comment by pnmueller** [comment]
As for the target dummy: I was using the one in Warspear. I also did some tests on my druid and all dummies outside of my garrison and inside Warspear were taking the exact same damage that also matched my calculation of what it should be. That means either the armour doesn't match or there's something else entirely going on. I'd assume it's the armour because I remember that Blizzard has used two different armour values for raid bosses for a long time (e.g. Brutallus had higher armour than Kil'jaeden back in Sunwell).

**Comment by pnmueller** [comment]
I'm not sure if I should open another issue or just post it here but here's a few more smaller issues I found after pulling the latest version:
- the Arms module doesn't seem to properly account for the ring granting you free 40 rage executes and instead only grants you 10 rage executes (tested on beta to confirm it works like this)
- the Fury_T19M.simc profile has the ID of the legendary ring but name, bonus ID, etc. of another one, just a small inconsistency that should probably be fixed
- the Arms APL doesn't react to the legendary ring proc (which it probably should once its damage is fixed) - a good guess would be to put it directly below MS or directly below the shattered defense lines in both partial APLs
- Anger Management now sims significantly higher for both the P and M Arms profiles compared to Opportunity Strikes which is taken in those profiles

**Comment by Collisionc** [comment]
Here is fine to post any issues, I'll sort them out.

**Comment by Collisionc** [comment]
> the Arms module doesn't seem to properly account for the ring granting you free 40 rage executes and instead only grants you 10 rage executes (tested on beta to confirm it works like this)

When I wrote that code, I assumed they were going to treat it like they did sudden death as they hadn't added legendarys to the flask vendors when I wrote it. I'll fix it.

> the Fury_T19M.simc profile has the ID of the legendary ring but name, bonus ID, etc. of another one, just a small inconsistency that should probably be fixed

Yeah, I hate making profiles. :D 

> the Arms APL doesn't react to the legendary ring proc (which it probably should once its damage is fixed) - a good guess would be to put it directly below MS or directly below the shattered defense lines in both partial APLs

I need to work on fury and arms action lists some, if you have any examples I can definitely add them in.

> Anger Management now sims significantly higher for both the P and M Arms profiles compared to Opportunity Strikes which is taken in those profiles

Yeah, I fixed a really significant bug with opportunity strikes a week ago that reduced the value of it a lot. I'll make sure to change that. 

> As for the target dummy: I was using the one in Warspear. I also did some tests on my druid and all dummies outside of my garrison and inside Warspear were taking the exact same damage that also matched my calculation of what it should be. That means either the armour doesn't match or there's something else entirely going on. I'd assume it's the armour because I remember that Blizzard has used two different armour values for raid bosses for a long time (e.g. Brutallus had higher armour than Kil'jaeden back in Sunwell).

It's probably a multiplier that I'm missing, I'll double check all the passive damage increases.

**Comment by Collisionc** [comment]
In regards to the 5% damage, try the target dummys in orgrimmar. I know for a fact that the garrison dummies are bugged, and maybe the warspear ones are too.

**Comment by pnmueller** [comment]
The ones in orgrimmar are actually taking slightly less minimum damage than the simulator has for the whole simulation's minimum damage (I removed flasks, food and potions). At this point, I'm fairly sure this is entirely related to armour since all target dummies are taking the exact same damage from moonfire.

**Comment by pnmueller** [comment]
Going to close the issue as everything's been addressed as far as I'm aware. Thanks for the quick response :)
